### PR TITLE
chore: update fmt-sql config and version

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

upgrade sqlfuff and tweak its configuration.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,11 @@
+# sqlfluff defaults: https://docs.sqlfluff.com/en/2.1.4/configuration.html#default-configuration
+
 [tool.sqlfluff.core]
 templater = "jinja"
 sql_file_exts = ".sql,.sql.j2,.dml,.ddl"
-exclude_rules = "L034,layout.spacing"
+exclude_rules = ""
+max_line_length = 100
+output_line_length = 100
```
Comment (stoksc) [inline master/pyproject.toml:8]: my take
```suggestion
max_line_length = 140
output_line_length = 140
```

Comment (hamidzr) [reply master/pyproject.toml:8]: went off of what we have for go but sgtm

File: master/static/srv/get_raw_allocation.sql [hunk 2]
```diff
@@ -135,15 +136,16 @@ SELECT
     start_time,
     end_time,
     extract(
-            epoch
-            FROM
-                -- `*` computes the intersection of the two ranges.
-                upper(const.period * tstzrange(start_time, end_time)) - lower(const.period * tstzrange(start_time, end_time))
-        ) AS seconds
+        EPOCH
```
Comment (hamidzr) [inline master/static/srv/get_raw_allocation.sql:139]: is this okay?

Comment (stoksc) [reply master/static/srv/get_raw_allocation.sql:139]: yeah, pretty sure

**Review by stoksc — APPROVED** [review]
lgtm

**Pull request merged**